File: ruby/triangle/triangle.rb
class Triangle
  def initialize(sides)
    @sides = sides.sort
  end

  def equilateral?
    @sides.uniq.size == 1 && @sides[0] != 0
  end

  def isosceles?
    @sides[1] == @sides[2] && self.is_triangle
  end

  def scalene?
    @sides.uniq.size == 3 && self.is_triangle
  end

  def is_triangle()
    @sides[0] + @sides[1] >= @sides[2]
  end
end

File: python/triangle/triangle.py
def equilateral(sides):
    ok, uniqueness = checkTriangle(sides)
    return uniqueness == 1


def isosceles(sides):
    ok, uniqueness = checkTriangle(sides)
    return ok and uniqueness < 3


def scalene(sides):
    ok, uniqueness = checkTriangle(sides)
    return ok and uniqueness == 3


def checkTriangle(sides):
    if 0 in sides:
        return False, -1

    sides.sort()
    return sides[0] + sides[1] >= sides[2], len(set(sides))
